File: src/main/scala/resonantinduction/mechanical/process/mixer/MixerNode.java
package resonantinduction.mechanical.process.mixer;

import net.minecraft.tileentity.TileEntity;
import net.minecraftforge.common.util.ForgeDirection;
import resonantinduction.core.interfaces.IMechanicalNode;
import resonantinduction.mechanical.energy.grid.MechanicalNode;
import universalelectricity.api.core.grid.INode;
import universalelectricity.api.core.grid.INodeProvider;

import java.util.WeakHashMap;

public class MixerNode extends MechanicalNode
{

    public MixerNode(INodeProvider parent)
    {
        super(parent);
        maxDeltaAngle = Math.toRadians(45);
        sharePower = false;
    }

    @Override
    public boolean canConnect(ForgeDirection direction)
    {
        return direction == ForgeDirection.DOWN || direction == ForgeDirection.UP;
    }

    @Override
    public boolean inverseRotation(ForgeDirection dir)
    {
        return dir == ForgeDirection.DOWN;
    }

}

File: mechanic/src/main/scala/com/calclavia/edx/mechanical/mech/process/mixer/MixerNode.scala
package com.calclavia.edx.mechanical.mech.process.mixer

import com.calclavia.edx.mechanical.mech.grid.MechanicalComponent
import net.minecraftforge.common.util.ForgeDirection
import resonantengine.api.graph.INodeProvider

/**
 * Node designed just for the Mixer to use
 * @param parent - instance of TileMixer that will host this node, should never be null
 */
class MixerNode(parent: INodeProvider) extends MechanicalComponent(parent)
{
  override def canConnect(direction: ForgeDirection): Boolean =
  {
    return direction == ForgeDirection.DOWN || direction == ForgeDirection.UP
  }

  /*
  override def inverseRotation(dir: ForgeDirection): Boolean =
  {
      return dir == ForgeDirection.DOWN
  }*/
}